geomstats/geometry/graphspace.py

```diff
@@ -0,0 +1,258 @@
+"""Graph Space."""
+
+import geomstats.backend as gs
+from geomstats.geometry.matrices import Matrices, MatricesMetric
+
+
+class _GraphSpace:
+    r"""Class for the Graph Space.
+
+    Graph Space to analyse populations of labelled and unlabelled graphs.
+    The space is a quotient space obtained by applying the permutation
+    action of nodes to the space of adjacency matrices.
+
+    Points are represented by :math:`nodes \times nodes` adjacency matrices.
```

Comment:
Adjacency matrices can thus be non-symmetric, thus we are looking at directed weighted graphs? Should we specify this?

Reply:
Graph Space to analyse populations of labelled and unlabelled graphs.
    The space focuses on graphs with scalar euclidean attributes on nodes and edges, 
    with a finite number of nodes and both directed and undirected edges. 
    For undirected graphs, use symmeric adjacency matrices. The space is a quotient 
    space obtained by applying the permutation action of nodes to the space 
    of adjacency matrices.